src/ktopt_planning_context.cpp

```diff
@@ -75,10 +137,21 @@ void KTOptPlanningContext::solve(planning_interface::MotionPlanResponse& res)
   auto traj = trajopt.ReconstructTrajectory(result);
   const size_t num_pts = 101;  // TODO: should be sample time based instead
   const auto time_step = traj.end_time() / static_cast<double>(num_pts - 1);
+  res.trajectory = std::make_shared<robot_trajectory::RobotTrajectory>(
+    start_state.getRobotModel(),
+    group
+  );
+  res.trajectory->clear();
+  const auto& active_joints = res.trajectory->getGroup() ? res.trajectory->getGroup()->getActiveJointModels() : res.trajectory->getRobotModel()->getActiveJointModels();
+
+  // sanity check
+  assert(traj.rows() == active_joints.size());
   for (double t = 0.0; t <= traj.end_time(); t += time_step)
   {
     const auto val = traj.value(t);
-    // TODO: Put into the robot trajectory in the response.
+    const auto waypoint = std::make_shared<moveit::core::RobotState>(start_state);
+    set_joint_positions(val, active_joints, *waypoint);
```

Comment:
Doesn't ktop also create velocities & accelerations?

Reply:
Indeed, but that is implicit in the trajectory. So as long as you get the timesteps and the values right in the `MotionPlanResponse`, this should be taken care off. Or am I not understanding your question right.